tests/lean/2161.lean

```diff
@@ -0,0 +1,19 @@
+structure Foo where num : Nat deriving DecidableEq
+
+namespace Foo
+
+instance : OfNat Foo n := ⟨⟨n⟩⟩
+
+/-! # Example 1 -/
+
+@[irreducible] def mul (a b : Foo) : Foo :=
+  let d := Nat.gcd a.num 1
+  ⟨(a.num.div d) * (b.num.div d)⟩
+
+example : ((Foo.mul 4 1).mul 1).mul 1 = 4 := by decide
```

Comment:
This fails both before and after this PR, but it goes from failing slowly to failing quickly.  Is there a way to put a heartbeat limit on the test?

Reply:
That's a good idea, thanks

Reply:
I think you put the heartbeat limit on the wrong test!

Reply:
Yikes, thanks again. Now it's on both.